contrib/python/pypodman/pypodman/lib/parser_actions.py

```diff
@@ -94,21 +91,40 @@ def __init__(self,
             help=help,
             metavar=metavar)
 
-    def __call__(self, parser, namespace, values, option_string=None):
-        """Convert and Validate input."""
-        items = getattr(namespace, self.dest, None) or []
-        items = copy.copy(items)
+        if hasattr(signal, "Signals"):
 
-        choices = ('CMD', 'ENTRYPOINT', 'ENV', 'EXPOSE', 'LABEL', 'ONBUILD',
-                   'STOPSIGNAL', 'USER', 'VOLUME', 'WORKDIR')
+            def _signal_number(signame):
+                cooked = 'SIG{}'.format(signame)
+                try:
+                    return signal.Signals[cooked].value
+                except ValueError:
+                    pass
+        else:
 
-        opt, val = values.split('=', 1)
-        if opt not in choices:
-            parser.error('Option "{}" is not supported by argument "{}",'
-                         ' valid options are: {}'.format(
-                             opt, option_string, ', '.join(choices)))
-        items.append(values)
-        setattr(namespace, self.dest, items)
+            def _signal_number(signame):
+                cooked = 'SIG{}'.format(signame)
+                for n, v in sorted(signal.__dict__.items()):
+                    if n != cooked:
+                        continue
+                    if n.startswith("SIG") and not n.startswith("SIG_"):
```

Comment:
OK, I think I figured this out, and I think it's a bug; my suspicion is that it's a remnant from a reverse-logic version of this function (i.e. `_signal_name(signum)`). It is impossible for this line ever to be false, because `n` and `cooked` are string comparisons. Could I suggest this instead?
```python
    for n, v [....]:
        if n == cooked:
            return v
    return None
```

Reply:
I'll tweak this on future PR.  The string comparison works as expected.  But the following if is unneeded it I use your code.  Simple is better!  Thx for the review.